nautobot/dcim/filters.py

```diff
@@ -327,17 +337,55 @@ class Meta:
         model = Location
         fields = ["id", "name", "slug", "description"]
 
+    def generate_query__base_site(self, value):
+        """Helper method used by DynamicGroups and by _base_site() method."""
+        if value:
+            max_depth = max(loc.tree_depth for loc in Location.objects.with_tree_fields())
+            filter_name = "site__in"
+            params = Q(**{filter_name: value})
+            for _i in range(max_depth):
+                filter_name = f"parent__{filter_name}"
+                params |= Q(**{filter_name: value})
+            return params
+        return Q()
+
+    @extend_schema_field({"type": "string"})
+    def _base_site(self, queryset, name, value):
+        """FilterSet method for getting Locations that are assigned to a given Site(s) or descended from one that is."""
+        params = self.generate_query__base_site(value)
+        return queryset.filter(params)
+
     def generate_query__child_location_type(self, value):
+        """Helper method used by DynamicGroups and by _child_location_type() method."""
         if value:
             # Locations whose location type is a parent of value, or whose location type *is* value but can be nested
             return Q(location_type__children__in=value) | Q(location_type__in=value, location_type__nestable=True)
         return Q()
 
     @extend_schema_field({"type": "string"})
     def _child_location_type(self, queryset, name, value):
+        """FilterSet method for getting Locations that can have a child of the given LocationType(s)."""
         params = self.generate_query__child_location_type(value)
         return queryset.filter(params)
 
+    def generate_query__subtree(self, value):
+        """Helper method used by DynamicGroups and by _subtree() method."""
+        if value:
+            max_depth = max(loc.tree_depth for loc in Location.objects.with_tree_fields())
```

Comment:
```suggestion
            max_depth = Location.objects.with_tree_fields().extra(order_by=["-__tree.tree_depth"]).first().tree_depth
```
To change this to a `LIMIT 1` query